kube/services/jobs/config-fence-job.yaml

```diff
@@ -0,0 +1,139 @@
+
+                  # copy default to shared location so other image can save in
+                  # secret
+                  cp new-fence-user-config.yaml /mnt/shared/old-fence-user-config.yaml
+                else
+                  echo "using previous fence-user-config from secret..."
+                  cp /mnt/shared/old-fence-user-config.yaml new-fence-user-config.yaml
+                fi
+
+                echo "injecting creds.json into fence configuration..."
+                python /var/www/fence/config_helper.py -i /var/www/fence/creds.json -c new-fence-user-config.yaml
+
+                echo "updating fence configuration..."
+                cp new-fence-user-config.yaml /mnt/shared/new-fence-user-config.yaml
+              else
+                echo "ERROR: /var/www/fence/creds.json not found"
+              fi
+            else
+              echo "/mnt/shared/old-fence-user-config.yaml did not appear within timeout :-("
+            fi
+
+      - name: awshelper
+        image: quay.io/cdis/awshelper:master
+        imagePullPolicy: Always
+        volumeMounts:
+          - name: shared-data
+            mountPath: /mnt/shared
+        command: ["/bin/bash"]
+        args:
+          - "-c"
+          - |
+            # install json query helper if necessary ...
+            if ! which jq > /dev/null; then
+              export DEBIAN_FRONTEND=noninteractive
+              sudo -E apt install -y jq
+            fi
+
+            if kubectl get secrets/fence-user-config > /dev/null 2>&1; then
+                echo "extracting previous fence-user-config from secret..."
+                kubectl get secrets/fence-user-config -o json | jq -r '.data["fence-user-config.yaml"]' | base64 --decode > /mnt/shared/old-fence-user-config.yaml
+            else
+                echo "no previous fence-user-config found. will use default..."
+                touch /mnt/shared/old-fence-user-config.yaml
+            fi
+
+            # wait for other image to generate config
+            let count=0
+            while [[ ! -f /mnt/shared/new-fence-user-config.yaml && $count -lt 50 ]]; do
+              echo "waiting for /mnt/shared/new-fence-user-config.yaml"
+              sleep 2
+              let count=$count+1
+            done
+
+            # load old yaml file into secrets
+            if kubectl get secrets/fence-user-config > /dev/null 2>&1; then
+              kubectl delete secret fence-user-config
+            fi
+            echo "saving raw configuration into fence-user-config secret..."
+            kubectl create secret generic fence-user-config "--from-file=fence-user-config.yaml=/mnt/shared/old-fence-user-config.yaml"
+
+            if [[ -f /mnt/shared/new-fence-user-config.yaml ]]; then
+              echo "updating fence configuration and fence-config..."
+
+              if kubectl get secrets/fence-config > /dev/null 2>&1; then
+                kubectl delete secret fence-config
```

Comment:
At this point `fence-config` and `fence-user-config` secrets hold the same info? 
If a user edit `api_configs/fence_user_config.yaml` by hand, and deletes the `fence-user-config` secret, and runs `kube-setup-secrets` - that's a good flow, right?  Or does some data get lost?